### PR TITLE
refactoring pvc_clones test

### PR DESCRIPTION
**Pull request by Avilir**
This PR refractor the test for pvc_clones to reduce the memory consumption.

The original test consume more then 2GiB of memory, which cause the test to failed  , and now the test consume **only** ~400MiB

### CODE REVIEW

File: ocs_ci/ocs/cluster.py [hunk 1]
```diff
@@ -903,18 +903,20 @@ def delete_blockpool(self, pool_name):
         if pool_name == "ocs-storagecluster-cephblockpool":
             return
 
+        # To make the deletion time faster, delete the created pool brutally
+        patch = (
+            f"cephblockpool {pool_name} --type=merge -p "
+            '\'{"metadata":{"finalizers":null}}\''
+        )
```
Comment (ramkiperiy) [inline ocs_ci/ocs/cluster.py:910]: are we deleting this in ceph backend? If so do take care of left over PVCs and PVs.
If it's deleted in order then won't be a problem.

Comment (Avilir) [reply ocs_ci/ocs/cluster.py:910]: there will be no leftover, all PVC / PV are deleted before the pool is deleted.

File: tests/e2e/performance/csi_tests/test_pvc_clone_performance.py [hunk 2]
```diff
@@ -38,34 +136,194 @@ def setup(self):
...
+        except Exception as e:
+            logger.exception(f"The PVC was not created, exception [{str(e)}]")
+            raise PVCNotCreated("PVC did not reach BOUND state.")
+        # Wait for the PVC to be Bound
+        performance_lib.wait_for_resource_bulk_status(
+            "pvc", 1, self.namespace, constants.STATUS_BOUND, self.timeout, 5
+        )
+        logger.info(f"The PVC {self.pvc_obj.name} was created and in Bound state.")
+
+    def create_pod_and_wait_for_compleat(self, **kwargs):
```
[the remark this answers is not in the document]
Comment (Avilir) [reply tests/e2e/performance/csi_tests/test_pvc_clone_performance.py:251]: thanks, will be fixed.

File: ocs_ci/ocs/perftests.py [hunk 3]
```diff
@@ -890,3 +893,41 @@ def set_results_path_and_file(self, func_name):
 
         self.results_path = os.path.join("/", *self.results_path, func_name)
         self.results_file = os.path.join(self.results_path, "all_results.txt")
+
+    def create_fio_pod_yaml(self, pvc_size=1, filesize=0):
+        """
+        This function create a new performance pod yaml file, which will trigger
+        the FIO command on starting and getting into Compleat state when finish
```
Comment (ypersky1980) [inline ocs_ci/ocs/perftests.py:900]: Small spelling fix-  Complete instead of Compleat

**Review by ypersky1980 — DISMISSED** [review]
LGTM

File: ocs_ci/ocs/cluster.py [hunk 4]
```diff
@@ -903,18 +903,20 @@ def delete_blockpool(self, pool_name):
...
             logger.warning(f"BlockPoool {pool_name} couldnt delete")
             logger.info("Try to force delete it")
-            patch = (
-                f"cephblockpool {pool_name} --type=merge -p "
-                '\'{"metadata":{"finalizers":null}}\''
-            )
-            self.RBD.exec_oc_cmd(f"patch {patch}")
-        self.RBD.wait_for_delete(resource_name=pool_name)
+        # Wait for 30 seconds before brutally delete the bool.
+        time.sleep(30)
```
Comment (OdedViner) [inline ocs_ci/ocs/cluster.py:918]: Can we use a generator?

Comment (OdedViner) [reply ocs_ci/ocs/cluster.py:918]: Discussed offline

**Review by OdedViner — DISMISSED** [review]

**Comment by Avilir** [comment]
the test was verified locally

**Review by ypersky1980 — APPROVED** [review]
LGTM

**Review by ramkiperiy — APPROVED** [review]

**Review by pintojoy — APPROVED** [review]
LGTM

**Pull request merged**